docs/guides/deployment-patterns.md

```diff
@@ -0,0 +1,170 @@
+    url = 'https://api.github.com/repos/PrefectHQ/prefect'
+    api_response = httpx.get(url)
+    if api_response.status_code == 200:
+        repo_info = api_response.json()
+        stars = repo_info['stargazers_count']
+        forks = repo_info['forks_count']
+        contributors_url = repo_info['contributors_url']
+        average_commits = calculate_average_commits(contributors_url)
+        print(f"PrefectHQ/prefect repository statistics 🤓:")
+        print(f"Stars 🌠 : {stars}")
+        print(f"Forks 🍴 : {forks}")
+        print(f"Average commits per contributor 💌 : {average_commits:.2f}")
+    else:
+        raise Exception('Failed to fetch repository information.')
+    
+@flow()
+def calculate_average_commits(contributors_url):
+    response = httpx.get(contributors_url)
+    if response.status_code == 200:
+        contributors = len(response.json())
+    else:
+        raise Exception('Failed to fetch contributors.')      
+    commits_url = f'https://api.github.com/repos/PrefectHQ/prefect/stats/contributors'
+    response = httpx.get(commits_url)
+    if response.status_code == 200:
+        commit_data = response.json()
+        total_commits = sum(c['total'] for c in commit_data)
+        average_commits = total_commits / contributors
+        return average_commits
+    else:
+        raise Exception('Failed to fetch commit information.')
+
+if __name__ == "__main__":
+    calculate_average_commits()
+```
+
+### Flow of deployments
+![Deployments diagram](img/guides/deployment.png)
+
+Prefect flows can also start a run of another flow through a deployment—a specification that associates a flow with a particular infrastructure. In this case, the deployed flow isn’t so much “part of” the initial flow, as it is “called by” that flow. Once a flow has called another flow, it can wait for it to complete, or it can simply start the deployed flow and proceed with the rest of its tasks. A deployed flow can run on separate infrastructure from the initial flow. Our community calls this the orchestrator pattern. Prefect users can think of and use deployed flows much like an external web service.
+
+This pattern is most useful when you want both conceptual and execution separation. The conceptual complexity is about the same as using a subflow, but the execution complexity is greater, because there’s separate infrastructure and processes to think about. Still, the complexity can be worth it, particularly when certain tasks need a certain type of infrastructure, like GPUs.
+
+TODO: Test flow name
+
+```python
+from my_project.flows import calculate_average_commits
+from prefect.deployments import Deployment
+
+@flow
+def build_deployments(name):
+    deployment = Deployment.build_from_flow(
+    flow=name,
+    name=f"{name}-deployment", 
+    version=1, 
+    work_queue_name="default",
+    work_pool_name="default-agent-pool",
+)
+    deployment.apply()
+
+if __name__ == "__main__":
+    build_deployments(calculate_average_commits)
+```
```

Comment:
im not sure this is what we'd want to be showing here.

firstly, I would expect the use of `run_deployment` here for a "Flow of deployments"

second, with the beta tag coming off the `prefect deploy` + prefect.yaml deployment UX today, I'd think we'd want to avoid encouraging the use of `build_from_flow` (until some point in the future when it works nicely with workers and pull steps).

also
>  In this case, the deployed flow isn’t so much “part of” the initial flow, as it is “called by” that flow.

I'm not sure if I agree with this, a flow called by a another flow seems to be directly a part of the parent flow

Reply:
I can get that new deployment pattern update in! Was referencing how to build [deployments from an object,](https://docs.prefect.io/2.10.21/concepts/deployments/?h=deployment#create-a-deployment-from-a-python-object) but I can recommend the new deployment pattern instead

Reply:
my main point is that `run_deployment` should be used here to run a deployment, not creating a deployemnt